lib/camp.js

```diff
@@ -172,6 +173,22 @@ function getCompressedStream(ask) {
   return res;
 }
 
+// Concatenate an array of streams into a single stream.
+function concatStreams(array) {
+  var concat = new stream.PassThrough();
+
+  function pipe(i) {
+    var end = i === array.length - 1;
+    array[i].pipe(concat, {end: end});
+    if (!end) {
```

Comment:
Also, there was no infinite loop:
- If array is `[stream]`, `pipe(0)` will be called, `end` will be `true`, no recursion.
- If array is `[]`, `pipe(0)` will be called, `end` will be `false`, `array[i].pipe` will throw, no recursion.